cache/models/samples.js

```diff
@@ -758,23 +766,27 @@ module.exports = {
       reqBody[sampFields.UPD_AT] = dateNow;
       reqBody[sampFields.CREATED_AT] = dateNow;
 
-      const cmds = [];
-      cmds.push(redisOps.addAspectInSubjSetCmd(
-        subject.absolutePath, aspectObj.name)
-      );
       cmds.push(redisOps.addKeyToIndexCmd(sampleType, sampleName));
       cmds.push(redisOps.setHashMultiCmd(sampleType, sampleName, reqBody));
-      return redisOps.executeBatchCmds(cmds);
+
+      return redisOps.getAspectNamesInSubject(subject.absolutePath);
     })
-    .then(() => redisOps.getHashPromise(sampleType, sampleName))
-    .then((updatedSamp) => {
-      const sampleRes = sampleStore.arrayStringsToJson(
-        updatedSamp, constants.fieldsToStringify.sample
-      );
+    .then((aspectsStr) => {
 
-      // aspect is not attached to match existing behaviour
-      return sampleRes;
-    });
+      // add aspect to the set of aspects attached to the subject
+      const aspectNames = JSON.parse(aspectsStr);
+      if (aspectNames.indexOf(aspectObj.name.toLowerCase()) < 0) {
+        aspectNames.push(aspectObj.name);
+      }
+
+      return redisClient.hmsetAsync(subjKey, {
+        subjectId: subject.id,
+        aspects: JSON.stringify(aspectNames),
+      });
+    })
+    .then(() => redisOps.executeBatchCmds(cmds))
+    .then(() => redisOps.getHashPromise(sampleType, sampleName))
+    .then((updatedSamp) => cleanAddAspectToSample(updatedSamp, aspectObj));
```

Comment:
good catch. there isn't an aspect object attached to sample, in the current implementation of API. Will remove